resources/building-mobile-extensions/event-listeners.md

```diff
@@ -166,7 +117,66 @@ public class MyExtension extends Extension {
 }
 ```
 
-## What can you do in your event handler?
+{% endtab %}
+
+{% tab title="Objective-C" %}
+
+### **iOS**
+
+1. In Xcode, create a new file from the `Cocoa Touch Class` template and save it in your project.
+2. Name your class `MyExtensionListener`, and it should be a subclass to the `ACPExtensionListener` class.
+
+   The `MyExtensionListener.m` file will contain your extension interface declaration and will import `ACPExtensionListener.h`. In the example below, the methods that are available for overriding are also displayed:
+
+**MyExtensionListener.h**
+
+```objective-c
+#import <ACPCore_iOS/ACPExtensionListener.h>
+#import <ACPCore_iOS/ACPExtensionEvent.h>
+
+@interface MyExtensionListener : ACPExtensionListener
+    - (void) hear:(ACPExtensionEvent *)event;
+@end
+```
+
+1. At a minimum, you must provide an implementation for the hear method.  
+
+**MyExtensionListener.m**
+
+```objective-c
+#import "MyExtensionListener.h"
+#import "MyExtension.h"
+
+@implementation MyExtensionListener 
+    - (void) hear:(ACPExtensionEvent *)event {
+        MyExtension* parentExtension = [self getParentExtension];
+        if (parentExtension == nil) {
+            NSLog(@"The parent extension was nil, skipping event");
+            return;
+        }
+
+        [parentExtension handleEvent:event];
+}
+
+/**
+ * Returns the parent extension that owns this listener
+ */
+- (MyExtension*) getParentExtension {
+    MyExtension* parentExtension = nil;
+    if ([[self extension] isKindOfClass:MyExtension.class]) {
+        parentExtension = (MyExtension*) [self extension];
+    }
+
+    return parentExtension;
+}
+
+@end
+```
+
+{% endtab %}
+{% endtabs %}
+
+### What can you do in your event handler?
 
 Your listener has a reference to the parent extension that registered it. You can use this to centralize logic into your extension class and call into it from your listener. This also means you have access to the extension services API \(`ACPExtensionApi` on iOS and `ExtensionApi` on Android\) provided to the extension. This will allow you to manage your shared states or register additional listeners. You also have access to the core SDK \(`ACPCore`\(iOS\) or `MobileCore` \(Android\). This will allow you to dispatch events and receive responses.
```

Comment:
In the second sentence, please change "this" to "this reference" to make it clearer.